### ISSUE TITLE
Correctly blit last row of ncvisuals with odd row counts

### ISSUE THREAD

**Issue by dankamongmen** [dankamongmen/notcurses#566]
While working on #557, I noticed that `tria_blit()` was barfing on input ncvisuals with odd numbers of rows:

```
ROWS: 39 STRIDE: 208 (52) COLS: 52
MADE INITIAL ONE 20/52
==3750524== Invalid read of size 1
==3750524==    at 0x484E11A: tria_blit (blit.c:41)
==3750524==    by 0x484E4E4: ncblit_rgba (blit.c:89)
==3750524==    by 0x486BF5D: ncvisual_render (visual.cpp:649)
==3750524==    by 0x1A9E15: _DOCTEST_ANON_FUNC_2() (rotate.cpp:114)
==3750524==    by 0x14DF4D: doctest::Context::run() (doctest.h:5863)
==3750524==    by 0x14EBC4: main (main.cpp:105)
==3750524==  Address 0x73221c3 is 3 bytes after a block of size 8,112 alloc'd
==3750524==    at 0x483677F: malloc (vg_replace_malloc.c:309)
==3750524==    by 0x486A7AD: memdup (internal.h:647)
==3750524==    by 0x486AFC5: ncvisual_from_rgba (visual.cpp:218)
==3750524==    by 0x1A9CC5: _DOCTEST_ANON_FUNC_2() (rotate.cpp:112)
==3750524==    by 0x14DF4D: doctest::Context::run() (doctest.h:5863)
==3750524==    by 0x14EBC4: main (main.cpp:105)
==3750524==
==3750524== Conditional jump or move depends on uninitialised value(s)
==3750524==    at 0x484DF9B: ffmpeg_trans_p (blit.c:7)
==3750524==    by 0x484E12C: tria_blit (blit.c:41)
==3750524==    by 0x484E4E4: ncblit_rgba (blit.c:89)
==3750524==    by 0x486BF5D: ncvisual_render (visual.cpp:649)
==3750524==    by 0x1A9E15: _DOCTEST_ANON_FUNC_2() (rotate.cpp:114)
==3750524==    by 0x14DF4D: doctest::Context::run() (doctest.h:5863)
==3750524==    by 0x14EBC4: main (main.cpp:105)
==3750524==
```

I think this is because we're not setting up an empty row at the bottom. ncvisuals ought always end up with an even number of rows, since we map two rows to each display row. I don't know if this applies to the main path, but it definitely applies here, and I think it applies to the former, and further explains the crud we sometimes see at the very bottom of e.g. `outro`.

tl;dr: every `ncvisual->dstheight` ought be even. If we get an odd number of rows, populate the last row with all 0x00000000 (no color, wholly transparent).

**Comment by dankamongmen** [comment]
I fixed it for the `ncvisual_from_{rgba, bgrx}()` cases. It can't happen via `ncplane_visual_open()`:

```
  ncv->dstheight *= 2;
```

same with `ncvisual_from_file()`, for the same reason (assigned twice an integer). And I can't reproduce the crud. So I think we're good there. I still think we ought handle this in `tria_blit()`.

**Comment by dankamongmen** [comment]
Yeah, actually, it ought *only* be done in `tria_blit()`, and *not* in e.g. `ncvisual_from_rgba()`. If we do the latter (as we have now), and rotate it, we rotate over this crap empty row. Dumb! Remove that code and instead handle it only in `tria_blit()`.

**Comment by dankamongmen** [comment]
Done. I removed the special cases and handle this in `tria_blit()` now.
